src/phoenix/datasets/evaluators/utils.py

```diff
@@ -78,19 +68,26 @@ def create_evaluator(
     kind: Union[str, AnnotatorKind] = AnnotatorKind.CODE,
     name: Optional[str] = None,
     scorer: Optional[Callable[[Any], EvaluationResult]] = None,
-) -> Callable[[EvaluatorCallable], ExperimentEvaluator]:
+) -> Callable[[Callable[..., Any]], "Evaluator"]:
     if scorer is None:
         scorer = _default_eval_scorer
 
     if isinstance(kind, str):
         kind = AnnotatorKind(kind.upper())
 
-    def wrapper(func: EvaluatorCallable, name: Optional[str] = name) -> ExperimentEvaluator:
-        if name is None:
-            name = func.__name__
+    def wrapper(func: Callable[..., Any]) -> Evaluator:
+        nonlocal name
+        if not name:
+            if hasattr(func, "__self__"):
+                name = func.__self__.__class__.__name__
+            elif hasattr(func, "__name__"):
+                name = func.__name__
+            else:
+                name = str(func)
```

Comment:
Is this enabling us to decorate classes as well as functions?

Reply:
No, but sometimes a function doesn't have the __name__ attribute. An example is the output of partial(), at least when I tested on 3.8

Reply:
Also the function could be a method of a class, in which case the class name may be more informative, so that's what I took instead.